pkg/server/server.go

```diff
@@ -309,3 +324,25 @@ func mapConditions(conditions []metav1.Condition) []*pb.Condition {
 
 	return out
 }
+
+var BadErrProvider = errors.New("wrong provider name")
+
+// Authenticate generates and returns a jwt token using git provider name and git provider token
+func (s *applicationServer) Authenticate(_ context.Context, msg *pb.AuthenticateRequest) (*pb.AuthenticateResponse, error) {
+
+	if string(gitproviders.GitProviderGitHub) != msg.ProviderName &&
```

Comment:
I'm not sure there's a lot of value in the [`GitProviderName`](https://github.com/weaveworks/weave-gitops/blob/cf1654f34912bae653be91df69d1e3d855624a90/pkg/gitproviders/factory.go#L15) type here as you can see it complicates comparisons.

I'd probably switch back to a `string` which is what the Go stdlib does https://pkg.go.dev/net/http#pkg-constants